packages/webapp/src/components/FilterPage/index.tsx

```diff
@@ -75,16 +73,16 @@ const PureFilterPage = ({
         filters={filters}
         onChange={(filterKey, filterState) => {
           if (filterKey === 'DATE_RANGE') {
-            setTempFilter({
-              ...tempFilter,
-              ...(filterState.fromDate && { FROM_DATE: filterState.fromDate }),
-              ...(filterState.toDate && { TO_DATE: filterState.toDate }),
-            });
+            setTempFilter((prevTempFilter) => ({
```

Comment:
We need to retrieve the previous state in the set state call instead of using props because `setTempFilter` may be called multiple times when clear filters is clicked. The value of the state passed in from props will be stale since react batch updates the state.